src/common/bit_util.h

```
@@ -45,6 +45,12 @@ template <typename T>
     return static_cast<u32>(log2_f + static_cast<u64>((value ^ (1ULL << log2_f)) != 0ULL));
 }
 
+template <typename T>
+requires std::is_integral_v<T>
+[[nodiscard]] constexpr bool IsPow2(T value) {
+    return std::has_single_bit(static_cast<std::make_unsigned_t<T>>(value));
```

I'm not particularly sure it's wise to allow passing in signed values in this function and silently convert them to unsigned vaalues, given this'll give incorrect results for negative values, which is why even the stdlib function doesn't allow passing in signed types

---

Hmm, @BreadFish64 suggested this one. Would the previous solution which tests for positivity be a better solution instead of using `std::has_single_bit`

---

I would just constrain the function to unsigned values and any cases that make use of signed values can explicitly cast. Behaviorally, the functionality being tested for literally only makes sense with unsigned values. Implicitly casting to unsigned can hide bugs in call-sites.

---

The other method was `value > 0 && (value & (value - 1)) == 0`, whereas `std::has_single_bit` is functionally equivalent to `value != 0 && (value & (value - 1)) == 0`

---

The reason why I'd need support for signed integers is that Atmosphere implements an equivalent function using `value > 0 && (value & (value - 1)) == 0`, accepting signed integers.